cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs

```diff
@@ -220,3 +226,19 @@ runGovernanceRegistrationCertificateCmd anyReg outfp =
             . writeLazyByteStringFile outfp
             $ conwayEraOnwardsConstraints cOnwards
             $ textEnvelopeToJSON description registrationCert
+
+runGovernanceDRepMetadataHashCmd ::
+  DRepMetadataFile In
+  -> Maybe (File () Out)
+  -> ExceptT GovernanceCmdError IO ()
+runGovernanceDRepMetadataHashCmd drepMDPath mOutFile = do
+  metadataBytes <- firstExceptT ReadFileError $ newExceptT (readByteStringFile drepMDPath)
+  (_metadata, metadataHash) <-
+    firstExceptT GovernanceCmdDRepMetadataValidationError
+     . hoistEither
+     $ validateAndHashDRepMetadata metadataBytes
+  case mOutFile of
+    Nothing -> liftIO $ BS.putStrLn (serialiseToRawBytesHex metadataHash)
+    Just (File fpath) ->
+      handleIOExceptT (WriteFileError . FileIOError fpath)
+        $ BS.writeFile fpath (serialiseToRawBytesHex metadataHash)
```

Comment:
We have functions like `writeByteStringOutput` that case on the `Maybe` for you.